### PR TITLE
refactor & improve WorldEntity system

### PR DESCRIPTION
**Pull request by matrix-marker**
- [x] allows easier subclassing & spawning
- [x] split WorldManager up, creating WorldEntityManager
- [x] add method to move WorldEntities
- [x] add utility methods to create WorldEntities

### CODE REVIEW

File: src/main/java/core/world/DemoSpawners.java [hunk 1]
```diff
@@ -8,77 +9,54 @@
 
 public class DemoSpawners {
 
-    public static EntitySpawner createPlayerSpawner() {
-        EntitySpawner spawner = new EntitySpawner();
+    private static Spawner<WorldEntityWithSprite> demoSpawnerUtility(Vector2 position, Vector2 size, Sprite sprite) {
```
Comment (mdawess) [inline src/main/java/core/world/DemoSpawners.java]: Much much cleaner, great work!

File: src/main/java/core/world/WorldEntityWithSprite.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+package core.world;
+
+import com.badlogic.gdx.graphics.g2d.Sprite;
+import com.badlogic.gdx.math.Vector2;
+import com.badlogic.gdx.physics.box2d.BodyDef;
+import com.badlogic.gdx.physics.box2d.FixtureDef;
+
+public class WorldEntityWithSprite extends WorldEntity {
```
Comment (mdawess) [inline src/main/java/core/world/WorldEntityWithSprite.java:8]: Don't know if this is necessary as keeping it within the WorldEntity doesn't add too much bulk

Comment (matrix-marker) [reply src/main/java/core/world/WorldEntityWithSprite.java:8]: many WorldEntity objects need no sprite

File: src/main/java/core/world/DemoSpawners.java (hunk 1 above)
Comment (matrix-marker) [reply src/main/java/core/world/DemoSpawners.java]: thanks !

**Pull request merged**